### PR TITLE
Add No IC (0.Download, 1.Split, 2.Train)

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, I add no IC mitocheck_data processing for 0.Download, 1.Split, 2.Train modules. This involves refactoring most of the the Jupyter Notebooks and the Readmes for these modules.

### CODE REVIEW

File: 0.download_data/README.md [hunk 1]
```diff
@@ -9,15 +9,12 @@ Complete instructions for data download and preprocessing can be found at: https
...
-The version of mitocheck_data used is specified by the hash corresponding to a current commit.
-The current hash being used is `e1f86cd007657f8247310b78df92891b22e51621` which corresponds to [mitocheck_data/e1f86cd](https://github.com/WayScience/mitocheck_data/tree/e1f86cd007657f8247310b78df92891b22e51621).
-The `hash` variable can be set in [download_data.ipynb](download_data.ipynb) to change which version of mitocheck_data is being accessed.
+We specify the path to each set of `mitocheck_data` with `labeled_data_paths` in [download_data.ipynb](download_data.ipynb).
 
 ### Data Preview
 
 The labeled dataset includes CellProfiler (CP) and DeepProfiler (DP) features as well as metadata (location, perturbation, etc) for cells from the original MitoCheck project.
-The breakdown of cell counts by phenotypic class (as labeled manually by MitoCheck) is as follows:
+The breakdown of cell counts by phenotypic class for `ic` dataset (as labeled manually by MitoCheck) is as follows:
```
Comment (MikeLippincott) [inline 0.download_data/README.md]: consider defining `ic` dataset?

File: 0.download_data/scripts/nbconverted/download_data.py [hunk 2]
```diff
@@ -10,38 +10,57 @@
...
 # In[2]:
 
 
-labeled_data_hash = "20369033f579dca1334cb2c58a1c6d532322f93e"
-labeled_data_url = f"https://raw.github.com/WayScience/mitocheck_data/{labeled_data_hash}/3.normalize_data/normalized_data/training_data.csv.gz"
+labeled_data_hash__ic = "20369033f579dca1334cb2c58a1c6d532322f93e"
+labeled_data_url__ic = f"https://raw.github.com/WayScience/mitocheck_data/{labeled_data_hash__ic}/3.normalize_data/normalized_data/training_data.csv.gz"
 
+# ADD PATH TO DATA FROM GITHUB
+# labeled_data_hash = ""
```
Comment (MikeLippincott) [inline 0.download_data/scripts/nbconverted/download_data.py:24]: consider removing if not in use

File: 0.download_data/scripts/nbconverted/download_data.py [hunk 3]
```diff
@@ -10,38 +10,57 @@
...
+labeled_data_url__no_ic = f"/home/roshankern/Desktop/Github/mitocheck_data/3.normalize_data/normalized_data__no_ic/training_data.csv.gz"
```
Comment (MikeLippincott) [inline 0.download_data/scripts/nbconverted/download_data.py:25]: consider  changing to a relative path and resolving the absolute path via 
`pathlib.Path(f"{realtive_path}").resolve()`

This so others can use this code reproducibly who on other workstations that do not have a `roshankern` user

File: 0.download_data/scripts/nbconverted/download_data.py [hunk 4]
```diff
@@ -10,38 +10,57 @@
...
 
-# ### View counts of each class in the labeled dataset
+
+# ## Preview data
 
 # In[4]:
 
 
-labeled_data["Mitocheck_Phenotypic_Class"].value_counts()
+labeled_data
```
Comment (MikeLippincott) [inline 0.download_data/scripts/nbconverted/download_data.py:57]: consider removing? or print to head

File: 1.split_data/scripts/nbconverted/split_data.py [hunk 5]
```diff
@@ -17,64 +17,70 @@
...
+    
+    # ratio of data to be used for testing (ex 0.15 = 15%)
+    test_ratio = 0.15
+
+    # get indexes of training and testing data
+    training_data, testing_data = train_test_split(
+        labeled_data,
+        test_size=test_ratio,
+        stratify=labeled_data[["Mitocheck_Phenotypic_Class"]],
+        random_state=1,
```
Comment (MikeLippincott) [inline 1.split_data/scripts/nbconverted/split_data.py:54]: consider changing to 0 to match the way lab convention

File: 2.train_model/README.md [hunk 6]
```diff
@@ -25,8 +25,19 @@ We search over the following parameters: `[0. , 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.
...
+model_types = ["final", "shuffled_baseline"]
+feature_types = ["CP", "DP", "CP_and_DP", "CP_zernike_only", "CP_areashape_only"]
+balance_types = ["balanced", "unbalanced"]
+dataset_types = ["ic", "no_ic"]
+```
+Here, these types represent:
+- Which version of features the model is trained on. For `shuffled_baseline`, each column of the feature data is shuffled independently to create a shuffled baseline for comparison.
+- Which features to use for trainining.
+- Whether or not to balance `class_weight` of each model when training.
+- Which `mitocheck_data` dataset to use for feature training. We have datasets extracted with and without illumination correction.
```
Comment (MikeLippincott) [inline 2.train_model/README.md]: I think this section could benefit from bullet points for each feature type and then an indentation below describing the feature type. Consider this change to organize the information more, ultimately up to you though.

**Review by MikeLippincott — APPROVED** [review]
Good PR, lots of modularity and cleanliness. Feel free to merge after addressing comments!

File: 0.download_data/scripts/nbconverted/download_data.py (hunk 2 above)
Comment (roshankern) [reply 0.download_data/scripts/nbconverted/download_data.py:24]: I will leave so we know where to clearly add this in the future.

File: 0.download_data/scripts/nbconverted/download_data.py (hunk 3 above)
Comment (roshankern) [reply 0.download_data/scripts/nbconverted/download_data.py:25]: I will leave this as is so it is clear where to change later. Unfortunately we are developing the PRs a little out of order, but this will be fixed before merging into main!

File: 0.download_data/scripts/nbconverted/download_data.py (hunk 4 above)
Comment (roshankern) [reply 0.download_data/scripts/nbconverted/download_data.py:57]: I prefer showing the data this way in Jupyter notebooks as this shows the shape of the dataframe and the first/last couple of rows so we can clearly see variation among rows.
